broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java

```diff
@@ -894,15 +894,19 @@ public boolean executeSendMessageBack(List<MessageExt> msgList, String brokerNam
     }
 
     private void initialTransaction() {
-        this.transactionalMessageService = ServiceProvider.loadClass(ServiceProvider.TRANSACTION_SERVICE_ID, TransactionalMessageService.class);
```

Comment:
IMO,The point of this is to effectively use the SPI mechanism so that users can extend their own implementation, so load this path